scripts/doc/gen-widget-doc.js

```diff
@@ -3,16 +3,31 @@ let dmd = require('dmd');
 let fs = require('fs');
 let util = require('util');
 let path = require('path');
+let Transform = require('stream').Transform;
 
 /* paths used by this script */
 let p = {
-  src: path.resolve(__dirname, '../../widgets/**/*.js'),
+  src: [path.resolve(__dirname, '../../widgets/**/*.js'), path.resolve(__dirname, '../../lib/InstantSearch.js')],
   json: path.resolve(__dirname, '../../source.json'),
   output: path.resolve(__dirname, '../../docs/_includes/widget-jsdoc/%s.md')
 };
 
+// convert kind='constructor' to kind='function'
+let ctorToFunction = new Transform({decodeStrings: false});
+ctorToFunction._transform = (chunk, encoding, done) => {
+  let data = eval(chunk.toString()); // eslint-disable-line
+  data.forEach(function(e) {
+    if (e.kind === 'constructor') {
+      e.kind = 'function';
+    }
+  });
+  done(null, JSON.stringify(data));
+};
```

Comment:
WFM as it's not crippling the underlying code (implementation of InstantSearch), maybe we could be able to override the type with a specific tag, dunno